lib/functions/failunsupportedassets.js

```diff
@@ -36,7 +48,8 @@ class FailUnsupportedAssets extends AsyncGeneratorFunction {
 
                 // initiate upload rejects these characters
                 const filename = transferAsset.target.filename;
-                if (filename.match(/[[\]{}&:\\?#|*%]/g)) {
+                const fileNameMatch = (this.options && this.options.supportSpecialCharacters) ? /[[\]{}:\\?|*%]/g : /[[\]{}&:\\?#|*%]/g;
```

Comment:
Since we have two sets of allowed characters, perhaps it would be easier and more extensible to define the "always allowed" and "special characters" in consts.

Reply:
Great idea - I moved these two regexes into the constants file.

Taylor